modules/aws/config.go

```diff
@@ -0,0 +1,16 @@
+package aws
+
+import (
+	"sync"
+)
+
+var once sync.Once
+var GlobalCustomEndpoints *map[string]string
```

Comment:
So, this still has the same issue of concurrency. That is, if multiple tests, running concurrently, want different endpoints, it's not really possible. That said, I think I can live with that, as _probably_ the common use case is to set it once for the entire test suite. However, this all needs to be explained clearly via comments on the function below.

Reply:
In this case is not a issue at all. As you mentioned the goal is just change the endpoint to local for instance the local endpoint to S3 will not change between tests. I don't know if GO have a mechanics to do a global setup for the tests suite

Reply:
I tested TestMain and work fine as expected

Reply:
> As you mentioned the goal is just change the endpoints to local, for instance the local endpoint to S3 will not change between tests

In your specific usage of this, sure. But Terratest is open source, and once we expose this as a public API, people will use it in a thousand different ways. So if this is not goroutine-safe, we need to mark that vary clearly in the API docs.

Reply:
Also, why is `GlobalCustomEndpoints` a pointer to a map rather than just a map? I guess it's because you're allowing the user to override it, but it seems safer to copy the keys/values...

Reply:
Actually, here's what we can do:

Instead of using `once.Do` and silently ignoring future calls to `SetAwsEndpointsOverrides`, why not set it the first time and then return an error if it's called more than once? That way, a user would least get notified if invalid usage (including concurrent usage).

Reply:
I will remove the pointe and once.Do.

I have additional comments about it. Maybe Imaging the scenario he have  n one test or call just a few custom endpoints:

var customLocalEndpointsOne = map[string]string{
	"apigateway":     "http://localhost:4567",
	"cloudformation": "http://localhost:4581"
}

aws.SetAwsEndpointsOverrides(customLocalEndpointsOne) 
and in a second call:

var customLocalEndpointsTwo = map[string]string{
	"route53":        "http://localhost:4580",
	"redshift":       "http://localhost:4577",
	"s3":             "http://localhost:4572",
}
aws.SetAwsEndpointsOverrides(customLocalEndpointsOne) 


what do you think about evaluate if the service already exist in the global GlobalCustomEndpoints and during the SetAwsEndpointsOverrides evaluate merge them or if already exist as you mentioned throw an exception? 

@brikis98 

Let me know your opinion about

Reply:
Well, if you have the `once.Do`, then in your example, the call to `aws.SetAwsEndpointsOverrides(customLocalEndpointsTwo)` gets silently ignored, which would be very confusing. My point is that if we only want `SetAwsEndpointsOverrides` to be called once, we should be vocal about it and return an error when it's called a second time.